Sources/Subprocess/IO/Output.swift

```diff
@@ -142,14 +142,12 @@ public struct BytesOutput: OutputProtocol {
     internal func captureOutput(
         from diskIO: consuming TrackedPlatformDiskIO?
     ) async throws -> [UInt8] {
-        #if os(Windows)
-        let result = try await diskIO?.fileDescriptor.read(upToLength: self.maxSize) ?? []
-        try diskIO?.safelyClose()
-        return result
-        #else
-        let result = try await diskIO!.dispatchIO.read(upToLength: self.maxSize)
+        let result = try await AsyncIO.shared.read(from: diskIO!, upTo: self.maxSize)
```

Comment:
If it's safe to force unwrap here, should this function be taking an optional?

Reply:
This is code smell though... if we never expect it to be nil then we should mark it that way. That's probably out of scope of this PR, but it's not the first time this question came up.

Reply:
I refactored the code to centralize "unwrapping" to one location and used `fatalError()` to make it clear it is a programmer error to have nil diskIO.